docusaurus/docs/iOS/client/state-layer/state-layer-overview.md

```diff
@@ -0,0 +1,229 @@
+
+```swift
+try await chat.loadMessages(with: MessagesPagination(pageSize: pageSize))
+let allMessages = chat.state.messages
+```
+
+## Types Representing the State
+
+The state layer has objects representing different use-cases which enables accessing and manipulating data the Stream API provides. These types follow an architecture where high level types have methods for interacting with the Stream API and the state counterpart keeps the currently loaded data. The state object is observable and is backed by the local persistent store.
+
+* ChannelList
+  * ChannelListState
+* Chat
+  * ChatState
+  * MessageState
+* ConnectedUser
+  * ConnectedUserState
+* MemberList
+  * MemberListState
+* MessageSearch
+  * MessageSearchState
+* ReactionList
+  * ReactionListState
+* UserList
+  * UserListState
+* UserSearch
+  * UserSearchState
+
+These objects are created through factory methods in `ChatClient`. Factory methods create instances of `ChannelList`, `Chat`, `ConnectedUser`, `MemberList`, `MessageSearch`, `ReactionList`, `UserList`,  and `UserSearch`. The observable state is accessed through `state` properties what return the respective state object. `ChannelList`, `Chat`, `MemberList`, `ReactionList`, and `UserList` have a `get()` method for fetching the default set of data from the Stream API.
+
+### ChannelList
+
+`ChannelList` and its `ChannelListState` enable querying a list of channels from the Stream API and provide an interface for loading channels in a paginated manner.
+
+```swift
+let query = ChannelListQuery(
+    filter: .and([
+        .equal(.type, to: .messaging), 
+        .containMembers(userIds: ["thierry"])
+    ]),
+    sort: [Sorting(key: .lastMessageAt, isAscending: false)],
+    pageSize: 10
+)
+let channelList = chatClient.makeChannelList(with: query)
+// local state of the channel list (state from previous sessions)
+var channels = channelList.state.channels
+// fetch the first page of channels from the Stream API
+try await channelList.get()
+// fetch more channels
+try await channelList.loadMoreChannels()
+// access all the fetched channels
+channels = channelList.state.channels
+```
+
+Since `ChannelListState` is observable, we can observe and react to channel list changes.
+
+```swift
+let cancellable = channelList.state
+    .$channels
+    .sink { channels in
+        // do something with channels
+    }
+```
+
+### Chat
+
+`Chat` and its `ChatState` represent the state of a channel. In addition, `Chat` has a method for retrieving an observable `MessageState` type which represents a single message and its observable state like reactions and replies.
+
+```swift
+let channelId = ChannelId(type: .messaging, id: "general")
+let chat = chatClient.makeChat(for: channelId)
+// local state of the channel (state from previous sessions)
+var messages = chat.state.messages
+var members = chat.state.members
+// fetch the latest state from the Stream API and subscribe to server-side events
```

Comment:
Maybe it is more clear like so, without getting into implementation details? 🤔 
```suggestion
// fetch the latest state from the Stream API and subscribe to data changes.
```
or 
`// fetch the latest state from the Stream API and subscribe to new web-socket events`
if we want the implementation details, but a bit more accurate